examples/bits.rs

```diff
@@ -1,98 +1,135 @@
+    // `bitvec` does know how to view.
+    let raw_number = tau.to_bits();
+    // `bitvec` can also view bits from left to right, with `BigEndian`.
+    let bits = raw_number.bits::<BigEndian>();
 
-    /// Returns the size in bytes of the array
-    pub fn len_bytes(&self) -> usize {
-        self.bytes.len()
+    // Since we are reading bits from left to right, the turtle should move from
+    // left to right also. Change the `* 2.0` to move the section horizontally.
+    let left_edge = -right_edge * 2.0;
+    turtle.set_x(left_edge);
+    // Walk from left to right
+    turtle.set_heading(0.0);
+
+    // The `&BitSlice` type acts just like `&[bool]`, so it comes with a
+    // `.chunks` method which divides it into smaller pieces. `bitvec` can take
+    // any number, not just multiples of 8, but 16 is a convenient number to
+    // look at. Try changing it to a different number, like 10, to see what
+    // happens!
+    for row in bits.chunks(16) {
+        // Each chunk produced is a smaller `&BitSlice`, just like
+        // `&[bool].chunks` produces smaller `&[bool]`s, so we can draw it.
+        draw_row(&mut turtle, row);
+        // After each row, the turtle has to go back to the left edge.
+        turtle.set_x(left_edge);
     }
+}
 
-    /// Set the bit at the given index to the given value
-    ///
-    /// Bits are set left to right, much like how an actual Vec might be indexed
-    ///
-    /// # Panics
-    /// Panics if index >= len()
-    pub fn set(&mut self, index: usize, value: bool) {
-        let byte_index = index / BITS;
-        let byte = &mut self.bytes[byte_index];
-        let bit_index = index % BITS;
-        // We want to make sure that bits are set left to right
-        // This depends on the platform's endianess
-        //TODO: Check this somehow
-        let bit_index = if cfg!(target_endian = "big") {
-            bit_index
-        } else {
-            BITS - 1 - bit_index
-        };
-        if value {
-            // Set the bit to 1 by performing a bitwise OR with each bit in the byte.
-            // `1 << bit_index` will make sure the bit_index bit is 1.
-            // 1 OR anything will always result in 1, so the bit will be set to 1.
-            *byte |= 1 << bit_index;
-        } else {
-            // Set the bit to 0 by performing a bitwise AND with each bit
-            // `!(1 << bit_index)` sets the bit_index bit to 0 and the other bits to 1.
-            // 1 AND x will always leave x as is.
-            // 0 AND x will set the bit to 0.
-            *byte &= !(1 << bit_index);
+/// Draw a row of bits on the screen.
+///
+/// This takes a reference to a turtle, which draws, and a reference to a slice
+/// of bits, which provides the data to draw.
+fn draw_row<C, T>(t: &mut Turtle, row: &BitSlice<C, T>)
+where C: Cursor, T: BitStore {
+    // `&BitSlice` can iterate over bits. It is just like `&[bool]`, and so it
+    // produces `&bool` for each loop.
+    for bit in row {
```

Comment:
nit: could you match on `&bit` instead of dereferencing `*bit` below?

Reply:
I'll be real, I forgot I could do that